[FEAT] Create Day4 Scene

eisyxx opened the pull request:
## ✅ 𝗖𝗵𝗲𝗰𝗸-𝗟𝗶𝘀𝘁
- merge할 브랜치의 위치를 확인해 주세요. (main❌/develop⭕️)
- 리뷰가 필요한 경우 리뷰어를 지정해 주세요.
- 리뷰는 PR이 올라오면 최대한 빠르게 진행합니다.
- P1 단계의 리뷰는 빠르게 확인 후 반영합니다.
- Approve된 PR은 assigner가 머지하고, 
수정 요청이 온 경우 수정 후 다시 push를 합니다.

## 📌 𝗜𝘀𝘀𝘂𝗲𝘀
- closed #32 

## 📎 𝗪𝗼𝗿𝗸 𝗗𝗲𝘀𝗰𝗿𝗶𝗽𝘁𝗶𝗼𝗻
- 스토리 진행 위한 대화 시스템 구현 완료
- 접근 제한 구역 설정
- ~`Assets/Script/RemoveNPCDialogueTrigger.cs`: 4일차 기본 스토리 진행 후 태주 사라지도록~
- `NPCPathMover.cs` 스크립트 할당하여 태주 이동 구현
- `TestDialogue.cs`에 npcMoveTriggerIndex 변수 추가하여 인스펙터에서 각 대화 트리거 오브젝트별로 NPC 이동을 실행할 문장 인덱스 설정하도록 수정, `DialogueManager.ShowDialogue()`에 인덱스를 전달하도록 구조 수정
- <img width="308" height="96" alt="스크린샷 2025-07-18 14 44 00" src="https://github.com/user-attachments/assets/4801ee90-ce9e-44be-8b75-8f6b754c776c" />
- npc 이동 후 대사 한번씩 더 나오는 문제 해결: `DialogueManager.cs`의 StartDialogueCoroutine() 로직 미세 조정
- <img width="451" height="157" alt="스크린샷 2025-07-18 14 44 30" src="https://github.com/user-attachments/assets/e7356ea3-55a3-420b-8dd8-6612535d3aba" />
- 카메라 배경색 변경, 대화창 텍스트 좌상단 정렬


## 📷 𝗦𝗰𝗿𝗲𝗲𝗻𝘀𝗵𝗼𝘁

https://github.com/user-attachments/assets/c8f05698-2d0e-4535-9f01-2a74a4a5e641


## 💬 𝗧𝗼 𝗥𝗲𝘃𝗶𝗲𝘄𝗲𝗿𝘀
- 세이브 시스템 및 출석부 디자인 나오면 퇴근 시퀀스 디벨롭 해야함
- 회의에서 접근 제한 구역 로직 통일되면 이것도 추후 보완 예정

eisyxx committed:
feat: 대사 작성

eisyxx committed:
fix: dialogue system

- chatbox 디자인 변경된 것 적용
- 글꼴 적용

- 두번째부터 캐릭터 sprite 이미지가 chatbox에 가려지던거 해결: order in layer 조절
- 아기들과 부딪히면 못 움직이게 되던 문제 해결: testdialogue 스크립트 삭제
- 캐릭터 스케일 조정

- Day4 씬 생성
- 카메라 에러 해결 중...

eisyxx committed:
Merge branch 'dev' into 32-feat-day4

eisyxx committed:
feat: day4 story

- Scene Templete 생성: 1st floor 씬을 기준으로 함. 추후 다른 씬들 만들 때 활용 예정
- Day4 씬 재생성 후 대사 추가

- 카메라 범위 조정으로 대화 시스템 애니메이션이 보이지 않는 문제가 있으나 추후 Dialogue Manager 수정하면 해결될 것으로 예상

eisyxx committed:
fix: fill sprite render

eisyxx committed:
Merge branch 'dev' into 32-feat-day4

eisyxx committed:
fix: chatbox reveal

- 대화창 안보이는 문제 해결: pixel per unit=1 설정
- 맵 수정: 4일차에 필요한 맵만 남기고 다 삭제, map transfer 재설정

eisyxx committed:
Merge branch 'dev' into 32-feat-day4

eisyxx committed:
fix: movingob, PM script

eisyxx committed:
chore: script 주석처리

eisyxx committed:
feat: RemoveNPCDialogueTrigger.cs

대화 후 태주 사라지도록

eisyxx committed:
chore: cam bgcolor

three-co1ors reviewed (approved):
대화 끝나고 태주 사라지는 것만 고치면 될 것 같아요! 수고하셨습니다~!

eisyxx commented:
그거 스크립트에 태주가 자리 비운다고 되어있어서 일부러 저렇게 한건데 좀 이상한가요...? 어떻게해야되지

sssuhha commented:
대화창 텍스트는 왼쪽 위 정렬로 부탁드려용

usuuhyn reviewed (changes requested):
Day3에 제가 한 것처럼 입구로 나가게끔 하면 될듯..? 
label, assignee 설정해주세요

three-co1ors committed:
중간 커밋 (Day1~5)

three-co1ors committed:
Collider 조절

중간 commit

three-co1ors committed:
중간 commit

eisyxx committed:
Merge branch 'dev' into 32-feat-day4

eisyxx committed:
fix: npc disappear

대화 후 태주 안 없어지게 수정
npc 태그 할당
대화창 텍스트 좌상단 배치

eisyxx committed:
feat: 대화 후 npc 이동

- npcPathMover 스크립트 할당
- TestDialogue.cs에 npcMoveTriggerIndex 추가 하여
Inspector에서 각 대화 트리거 오브젝트별로 NPC 이동을 실행할 문장 인덱스 설정 가능하도록 함
DialogueManager.ShowDialogue()에 인덱스를 전달하도록 구조 수정

eisyxx committed:
Merge remote-tracking branch 'origin/54-feat-day1-5-Modify' into 32-feat-day4
